Porting FxCop Rules to Roslyn.md

```diff
@@ -0,0 +1,66 @@
+
+## Analyzer packages
+
+The list of all the rules that ship in VS, along with certain other FxCop/Roslyn rules that we know of, is captured in the file [RulesInventory.csv](https://github.com/dotnet/roslyn-analyzers/blob/master/RulesInventory.csv) file (which, thanks to GitHub, is searchable). That file also contains our proposed factoring of the analyzers (in the "Proposed Analyzer" column, which perhaps might have been better named "Proposed Analyzer Package").
+
+### API analyzer packages
+
+There are rules about types in the following contract assemblies:
+
+* **System.Runtime.Analyzers** - This package already exists
+
+* **System.Runtime.InteropServices.Analyzers** - Analyzers about interop and marshalling. This package already exists.
+
+* **System.Security.Cryptography.Algorithms** - Analyzers with guidelines for crypto algorithms usage. This is a new package.
+ 
+* **System.Xml** - Contains analyzers for types dealing with XML across  the System.Xml.* contracts. This is a new package.
+
+* **Desktop.Analyzers** - Contains analyzers for APIs that are present in the desktop .NET Framework but not in the new .NET Core API set. Since the .NET framework isn't available in a piecemeal fashion, there's not much value in breaking this down further.
+
+* **Microsoft.CodeAnalysis.Analyzers** - Contains rules about using the Roslyn APIs correctly. This package already exists.
+
+### Theme-based analyzer packages
+
+* **Microsoft.ApiDesignGuidelines** - Contains guidelines for authoring libraries which contain public APIs. The advantage of factoring it out this way is that one could simply install this analyzer for projects that expose real public APIs, and not for executables and test projects, reducing noise significantly.
+ 
+* **Microsoft.Maintainability** - Contains rules that contains metrics-based and heuristics-based rules to assess complexity, maintainability, and readability.
+ 
+* **Text.Analyzers** - Contains rules that analyze code as text for spelling violations. 
+
+* **Roslyn.Internal.Analyzers** - Contains rules about some internal types in the Roslyn code base, meant as guidelines for Roslyn contributors as opposed to Roslyn consumers.
+
+## What to port?
+
+Aside from the factoring, the .csv file also has some information about telemetry that has been reported through VS about the number of violations and suppressions for many of the rules. We used that as one data point in identifying whether a rule was high value. Of course some of those numbers might have been reported long ago, so a subjective evaluation of the usefulness of a rule was needed. We have populated the "Port?" column of the spreadsheet with our decisions. (NOTE: To see that column, you'll need to scroll to the bottom of the page and scroll horizontally.)
```

Comment:
If you care to amplify on this, we examined both the frequency that rules fire and the frequency that their associated results are suppressed. We tended to favor rules that are not suppressed to a high degree. We did not throw out rules as a matter of course that fire infrequently: there are several useful checks which don't fire often but flag good issues and always indicate an issue. We also were critical of rules at the end of the 'fire frequency' spectrum, throwing out some checks that never/rarely fire (some of these related to compiler fixes that actually prevent older, bad MSIL patterns from occurring) and deprioritizing some checks that are extremely noisy (which argues for an improved analysis spec, which was out of scope for the exercise)